src/runtime/HalideRuntime.h

```diff
@@ -742,6 +742,24 @@ extern void halide_profiler_reset();
  * reset. Also happens at process exit. */
 extern void halide_profiler_report(void *user_context);
 
+/// \name halide_float16_t type and associated functions
+///{@
+/**
+ * This type represents the bits for the IEEE-754 2008 binary16 floating point
+ * format (also known as "half")
+ */
+typedef uint16_t halide_float16_t;
```

Comment:
Do you think it would be better if we skipped the typedef entirely, and just made the functions below take a uint16_t (and adjust the comments to explain that the bits will be interpreted as an IEE-754 float16)? That way there's no danger of someone adding two halide_float16_ts together and getting a nasty surprise.